Add YandexSamokatTest and MakeOrderYandexSamokatTest

VladimirSafronov opened the pull request.

VladimirSafronov committed:
Add YandexSamokatTest and MakeOrderYandexSamokatTest

pomegranate282922 commented on src/main/java/pages/CustomerInfo.java:
```diff
@@ -0,0 +1,119 @@
+
+  /**
+   * Метод заполняет поля для заказа
+   *
```
Отлично что есть джавадока

pomegranate282922 commented on src/test/java/BaseTest.java:
```diff
@@ -0,0 +1,33 @@
+    driver.manage().timeouts().pageLoadTimeout(30, TimeUnit.SECONDS);
+    driver.manage().timeouts().setScriptTimeout(30, TimeUnit.SECONDS);
+
+    driver.get("https://qa-scooter.praktikum-services.ru/");
```
Урлы лучше вынести как константу и переиспользовать 
Можно улучшить

VladimirSafronov replied:
то есть создать константу в классе BaseTest, а driver.get() вызывать в самом тесте с константой?

pomegranate282922 replied:
Лучше константы в отдельный класс положить, чтоб не привязываться логически к BaseTest

pomegranate282922 commented on src/test/java/MakeOrderYandexSamokatTest.java:
```diff
@@ -0,0 +1,54 @@
+        "Всплывающее окно о подтверждении заказа не появилось. Текст заказа: " + orderInfoText,
+        orderInfoText.startsWith("Заказ оформлен"));
+  }
+}
```
Нужно еще сделать тест второго флоу создания заказа: проверить что мы доходим до формы ввода данных заказа 
Нужно исправить

pomegranate282922 commented on src/main/java/pages/YandexSamokatMain.java:
```diff
@@ -0,0 +1,121 @@
+      String answer = getQuestionOrAnswerText(false, i);
+      System.out.println("Question: " + question + " Answer: " + answer);
+      questionsAnswers.put(question, answer);
+    }
```
Так не очень оптимально. На каждом тесте ( а их 8) ты пробегаешь по всем вопросам, хотя нас интересует только один. 
можно улучшить

pomegranate282922 commented on src/test/java/BaseTest.java:
```diff
@@ -0,0 +1,33 @@
+
+  @Before
+  public void before() {
+    System.setProperty("webdriver.chrome.driver", System.getenv("CHROME_DRIVER"));
```
Отлично что используется энв переменная

pomegranate282922 commented on src/test/java/MakeOrderYandexSamokatTest.java:
```diff
@@ -0,0 +1,54 @@
+    customerInfo.clickButtonNext();
+
+    OrderInfo orderInfo = new OrderInfo(driver);
+    orderInfo.inputOrderInfo();
```
Здесь надо значения передать через параметры, а не хардкодить в po- иначе po получается очень ограниченным по использованию.
Нужно исправить

pomegranate282922 reviewed (commented):
Работа хорошая! Применена параметризация, есть описание для джавадоки. 
замечания:
1. Добавить тест второго флоу 
2. Убрать хардкод из po 
3. Остальное по мелочи

VladimirSafronov committed:
Create Constants

 - develop parameterized inputOrderInfo()
 - optimize test openQuestionsBlockThanAnswersCorrect
 - create test clickLowerOrderButtonThenOpenOrderWindow

pomegranate282922 commented on src/test/java/MakeOrderYandexSamokatTest.java:
```diff
@@ -0,0 +1,81 @@
+  }
+
+  @Test
+  public void clickLowerOrderButtonThenOpenOrderWindow() {
```
Этот Тест повторится дважды, так как лежит в параметризованном классе. 
можно улучшить

pomegranate282922 reviewed (approved):
Ок, критичных замечаний нет - зачет!
